### PR TITLE
Convert rediscala test to scala

### PR DESCRIPTION
**Pull request by laurit**
Part of https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7195
Alternative to https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12553

### CODE REVIEW

File: instrumentation/rediscala-1.8/javaagent/src/test/scala/io/opentelemetry/javaagent/instrumentation/rediscala/RediscalaClientTest.scala [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+              span.hasName("parent").hasNoParent
+            }
+          },
+          new Consumer[SpanDataAssert] {
+            override def accept(span: SpanDataAssert): Unit = {
+              span
+                .hasName("SET")
+                .hasKind(CLIENT)
+                .hasParent(trace.getSpan(0))
+                .hasAttributesSatisfying(
```
Comment (jaydeluca) [inline instrumentation/rediscala-1.8/javaagent/src/test/scala/io/opentelemetry/javaagent/instrumentation/rediscala/RediscalaClientTest.scala:168]: is there a reason not to use `hasAttributesSatisfyingExactly` for these?

Comment (laurit) [reply instrumentation/rediscala-1.8/javaagent/src/test/scala/io/opentelemetry/javaagent/instrumentation/rediscala/RediscalaClientTest.scala]: Thanks, there isn't. Copy pasted it from the java conversion PR.

**Review by jaydeluca — APPROVED** [review]

**Review by steverao — APPROVED** [review]
BTW, if you can modify the hyperlink to https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12553#discussion_r1836100152 in PR description, it will be better for others to understand the background.

**Comment by trask** [comment]
good idea, I think we've discussed this way back that it's nice to write tests in the language of the library (e.g. kotlin)

**Comment by shalk** [comment]
> good idea, I think we've discussed this way back that it's nice to write tests in the language of the library (e.g. kotlin)

I see

**Pull request merged**